### PR TITLE
Sprint 8 fixes

### PR DESCRIPTION
**Pull request by Nishanthh**

### CODE REVIEW

File: src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts [hunk 1]
```diff
@@ -316,4 +318,32 @@ export class ConsultationQuestionnaireComponent implements OnInit, AfterViewInit
     });
   }
 
+  scrollToFirstInvalidControl() {
+    let errorElementFound;
+    const checkErrorElementExist = setInterval(() => {
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts:323]: @Nishanthh Bro why is this setInterval there?

File: src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts [hunk 2]
```diff
@@ -316,4 +318,32 @@ export class ConsultationQuestionnaireComponent implements OnInit, AfterViewInit
     });
   }
 
+  scrollToFirstInvalidControl() {
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts:321]: @Nishanthh Bro a similar function is already defined in modular.functions.ts. Can that function itself be reused here?

File: src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.html [hunk 3]
```diff
@@ -1,5 +1,56 @@
-<div class="response-index mt-4" ngResizable [rzHandles]="'s'"
-#responseIndex>
+<div class="paper-effect  mt-4 m-0-sm">
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.html:1]: @Nishanthh Bro is it possible to make this as a component since its being used both in consultation-response-text.component.html  and consultation-questionnaire.component.html?

File: src/app/shared/functions/modular.functions.ts [hunk 4]
```diff
@@ -197,3 +197,20 @@ export function isMobile() {
         return false;
     }
 }
+
+export const scrollToFirstError = (selector, element) => {
+    let errorElementFound;
+    const checkErrorElementExist = setInterval(() => {
```
Comment (avikt) [inline src/app/shared/functions/modular.functions.ts:203]: @Nishanthh Same question here- What's is the use of the setInterval here?

**Review by avikt — COMMENTED** [review]
Left a few comments and one change request

File: src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts (hunk 1 above)
Comment (Nishanthh) [reply src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts]: @avikt couldn't find the error element instantly, handled with afterviewchecked

File: src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts (hunk 2 above)
Comment (Nishanthh) [reply src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts]: @avikt yes, used the modular function now

File: src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.html (hunk 3 above)
Comment (Nishanthh) [reply src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.html:1]: @avikt yes bro, make into seperate component

File: src/app/shared/functions/modular.functions.ts (hunk 4 above)
Comment (Nishanthh) [reply src/app/shared/functions/modular.functions.ts]: @avikt couldn't find the error element instantly, handled with afterviewchecked

File: src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.html [hunk 5]
```diff
@@ -88,7 +39,7 @@
         </div>
     </div>
     </div>
-    <p class="error-msg" *ngIf="showError && !responseText">Please enter response text</p>
+    <p class="error-msg" *ngIf="showError && !responseText">Please enter a Response Text to submit the response</p>
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.html:42]:
```suggestion
    <p class="error-msg" *ngIf="showError && !responseText">Please enter your response</p>
```

File: src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts [hunk 6]
```diff
@@ -51,42 +62,41 @@ export class ConsultationResponseTextComponent implements OnInit, AfterViewCheck
...
 
   ngAfterViewChecked() {
     this.editIframe();
+    if (this.scrollToError) {
+      scrollToFirstError('.error-msg', this.el.nativeElement);
+      this.scrollToError = false;
+    }
   }
 
   createSatisfactionRating() {
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts:76]: @Nishanthh Can we name this `createResponse`?

File: src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts [hunk 7]
```diff
@@ -147,7 +156,6 @@ export class ConsultationResponseTextComponent implements OnInit, AfterViewCheck
 
   enableSubmitResponse(value) {
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts:157]: @Nishanthh Bro is this being used anymore?

File: src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts [hunk 8]
```diff
@@ -51,42 +62,41 @@ export class ConsultationResponseTextComponent implements OnInit, AfterViewCheck
     this.getCurrentUser();
     this.subscribeUseTheResponseText();
     this.getResponseText();
-    this.scrollToCreateResponse();
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts:54]: @Nishanthh Why is this removed?

File: src/app/modules/consultations/consultation-profile/feedback-modal/feedback-modal.component.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit, ViewChild, Output, EventEmitter, Input, OnDestroy } from '@angular/core';
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/feedback-modal/feedback-modal.component.ts:1]: @Nishanthh We can get rid of this component now, right?

**Review by avikt — COMMENTED** [review]
Left some feedback

File: src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts (hunk 8 above)
Comment (Nishanthh) [reply src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts:54]: @avikt if we dont have a response text, triggers scroll to error and reach responsecreate container otherwise it submits the responsetext input in the container. so scrolling to response container is not needed

File: src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts (hunk 7 above)
Comment (Nishanthh) [reply src/app/modules/consultations/consultation-profile/consultation-response-text/consultation-response-text.component.ts:157]: @avikt yes bro, function name has been renamed conventionally bro

**Review by avikt — APPROVED** [review]
